### DISCUSSION TITLE
Any thoughts on OpenFeature?

### DISCUSSION THREAD

**Discussion by Starefossen** [Unleash/unleash#3896] [General]
Any thoughts on support for the [OpenFeature](https://openfeature.dev/) the open standard for feature flag management. I saw Unleash and @ivarconr name on [Interested Parties](https://openfeature.dev/community/interested-parties/#interested-organizations), but not much else.

OpenFeature looks very much like what [OpenTelemetry](https://opentelemetry.io/) is about to become, I don't think Unleash should miss out on this 🚀

**Comment by pransh15** [comment]
Hey @Starefossen 👋 

Thanks for starting this discussion. We are monitoring the space and are interested in OpenFeature, but currently we do not have the bandwidth internally to pursue this. As we always do, we listen to our community and insightful ideas to understand how we can we provide value.

We met with someone from the OpenTelemetry working group at Devoxx Poland just last week to understand their setup and that does look interesting. 💡 

Perhaps, if your or someone from the community would like to take this up, we'd be happy to help kickstart the initiative!

**Reply by Starefossen** [reply]
Thanks for taking time to answer this, @pransh15 🙏 I thought that would be the case, great to hear that you are positive. I’ll dig around to get a feel of the effort required to get something working 😅

**Reply by pransh15** [reply]
That sounds great, @Starefossen. Looking forward to hearing from you again. 😸

**Comment by itzsaga** [comment]
Wanted to add on that I believe this will be table stakes in the future and was one of the first things I looked into when I learned we use Unleash at my current employer.

**Reply by rroik** [reply]
I agree, and I’d like to highlight that OpenFeature support is a crucial factor in choosing a feature flagging system. Having OpenFeature compatibility would make it much easier for teams to transition from other providers to Unleash without extensive rework. It also enables multi-platform support during migration, allowing us to implement feature flags consistently across diverse environments

**Reply by black-snow** [reply]
We're currently evaluating and I was rather shocked tbh that Unleash doesn't support it. Would've been my favourite but lack of OpenFeature support is almost a red flag atm.

**Comment by toddbaert** [comment]
Hey! I'm a maintainer over at OpenFeature! We'd love to see an Unleash provider (we've had frequent requests for them). We'd happily host them in our various [contrib-repos](https://github.com/search?q=org%3Aopen-feature+contrib&type=repositories&s=stars&o=desc), which house providers for other OSS projects and vendors, complete with CI and publishing. It looks like @liran2000 has already opened a [Java provider PR](https://github.com/open-feature/java-sdk-contrib/pull/424).

@pransh15 On the topic of OpenTelemetry integration... this is something we're quite interested in as well. We have some [built in extensions (hooks)](https://openfeature.dev/ecosystem?instant_search%5Bquery%5D=opentelemetry&instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Hook) to enable OTel-semantic conformant metrics and traces for feature-flag evaluations. They can essentially be wired up with a single line of code to any OpenFeature SDK.

We also helped develop the [OTel semantic-conventions](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/feature-flags/) for feature flags, which can really be leveraged by anyone, even without OpenFeature. I hope they might be generally useful to the Unleash project - we tried to keep them as generic as possible.

**Reply by ivarconr** [reply]
> Hey! I'm a maintainer over at OpenFeature! We'd love to see an Unleash provider (we've had frequent requests for them). We'd happily host them in our various [contrib-repos](https://github.com/search?q=org%3Aopen-feature+contrib&type=repositories&s=stars&o=desc), which house providers for other OSS projects and vendors, complete with CI and publishing. It looks like @liran2000 has already opened a https://github.com/open-feature/java-sdk-contrib/pull/424.

This is really great @toddbaert . Thanks for bringing this to our attention. Would the Unleash team be able to patch fixes themselves? (PS! we will never consider an SDK official if the team is not able to fix any problems for our users).

> We also helped develop the [OTel semantic-conventions](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/feature-flags/) for feature flags, which can really be leveraged by anyone, even without OpenFeature. I hope they might be generally useful to the Unleash project - we tried to keep them as generic as possible.

This is really cool. What is the process of having opinion on the spec itself? We love that there are standardization on this, but also see some improvements that can make it more explicit. E.g. "enabled" and "variant" are actually two different concepts in feature flagging, and merging them down to one attribute takes away some of the semantics.

**Comment by toddbaert** [comment]
> This is really great @toddbaert . Thanks for bringing this to our attention. Would the Unleash team be able to patch fixes themselves? (PS! we will never consider an SDK official if the team is not able to fix any problems for our users).

Our contrib repositories are "monorepos" that contain a bunch of integrations, extensions, etc. We have a `component-owners` file (see [this example](https://github.com/open-feature/java-sdk-contrib/blob/main/.github/component_owners.yml) in the java-contrib) that helps keep track of ownership of each sub-component. Anyone listed as a `component-owner` is automatically added as an assignee/reviewer for changes that impact that component, and their review is required for changes.
For components involving vendors or OSS tools, we generally add as many established maintainers from that project as are interested.

Note that you're also completely welcome to host any OpenFeature providers in your own organization if you'd like full control.

> This is really cool. What is the process of having opinion on the spec itself? We love that there are standardization on this, but also see some improvements that can make it more explicit. E.g. "enabled" and "variant" are actually two different concepts in feature flagging, and merging them down to one attribute takes away some of the semantics.

OpenFeature makes a distinction between "enablement" and "variants" as well. Generally though the specification doesn't _directly_ support flags that have a multipart payload with a boolean state as well as some other auxiliary variant (there _are_ ways to accomplish this with the OpenFeature API, but exactly which would be best I can't say with my limited knowledge of Unleash). I know some flag providers _do_ support this, but many also do not. This is the kind of thing that's difficult when designing an open standard API - we're trying to build something that is maximally compatible with tens of paradigms. That said, these sorts of discussions are exactly what we're all about. The spec is completely open source, and it's been built by getting consensus from as many vendors as possible. We'd certainly welcome PRs or issues from Unleashed folks. If you'd like to specifically discuss how the OpenFeature API could best be used to match Unleash's concepts, I'd be happy to meet about that or start a discussion.

**Reply by ivarconr** [reply]
Thanks for clarifying. 

> This is the kind of thing that's difficult when designing an open standard API - we're trying to build something that is maximally compatible with tens of paradigms.

I do understand that building one spec to match multiple vendors needs is challenging. 

> OpenFeature makes a distinction between "enablement" and "variants" as well. Generally though the specification doesn't directly support flags that have a multipart payload with a boolean state as well as some other auxiliary variant (there are ways to accomplish this with the OpenFeature API, but exactly which would be best I can't say with my limited knowledge of Unleash).

At unleash we believe feature management is a first class, and it is important to have a clear concise domain language. We believe a semantics beyond a single value is important. Based on our 9 year experience working with thousands of companies, we see that dynamic values are less common and often a miss-use of feature flags for static configuration which often leads to permanent flags with resistance to remove.

> The spec is completely open source, and it's been built by getting consensus from as many vendors as possible. We'd certainly welcome PRs or issues from Unleashed folks. If you'd like to specifically discuss how the OpenFeature API could best be used to match Unleash's concepts, I'd be happy to meet about that or start a discussion.

We would love to see Unleash work great with Open Feature.

**Comment by BrandonAtHoag** [comment]
Would love to see this too. I'm not including Unleash in my company's feature flag solution considerations specifically because there's no Javascript provider for Openfeature.

**Reply by gastonfournier** [reply]
Thanks @BrandonAtHoag for sharing your feedback! We understand the importance of having a JavaScript provider for OpenFeature but currently, we don't have the capacity to take this on. It'd be good if you can open an issue [here](https://github.com/open-feature/js-sdk-contrib/) requesting for an Unleash provider. We'd be happy to assist by providing feedback and validation if someone from the community can take the lead on this effort, as we did with some existing OpenFeature Unleash providers for other languages.

**Reply by gastonfournier** [reply]
Hi @BrandonAtHoag out of curiosity, what's the main reason for choosing OpenFeature? We have some ideas but it's always better to know from the source.

**Comment by lornest** [comment]
Revisiting this thread in 2025 - any sign of OpenFeature support on the roadmap team?

**Reply by gastonfournier** [reply]
Currently not in our focus, but there's been 3 community contributions making OpenFeature-compatible Unleash client available in Java, Javascript and Golang: https://openfeature.dev/ecosystem/?instant_search%5BrefinementList%5D%5Bvendor%5D%5B0%5D=Unleash

**Comment by kikihakiem** [comment]
https://github.com/open-feature/python-sdk-contrib/pull/301 I would love to hear your feedback

**Reply by gastonfournier** [reply]
Thanks! I took a look at it, and as soon as I have more time, I'll check your comments. I think I'll need to read a bit about OF protocol to give you a proper answer. Just FYI I'll try to answer as soon as I can.

**Comment by diktamxx** [comment]
+1

**Reply by gastonfournier** [reply]
Hi @diktamxx there are currently [3 languages supported for Unleash](https://openfeature.dev/ecosystem/?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Provider&instant_search%5BrefinementList%5D%5Bvendor%5D%5B0%5D=Unleash), mostly community-driven. We'd be happy to get more contributions cause right now it's not our main focus, but we're happy to see some movement there.

**Comment by gastonfournier** [comment]
There's a 4th language now, community-driven: https://pypi.org/project/openfeature-provider-unleash/

**Comment by askpt** [comment]
There is a new proposal for dotnet provider here community-driven: https://github.com/open-feature/dotnet-sdk-contrib/pull/652

**Comment by ivarconr** [comment]
[**2026, June**] We are currently scaling our team and now have the capacity to focus on OpenFeature providers for Unleash. We will begin by prioritizing missing providers for our most popular SDKs, followed by a comprehensive evaluation of all community-developed providers.

If you want to contribute, let us know.

**Reply by Drowze** [reply]
Excited for OpenFeature support!

We are heavily integrated with Datadog and, as we also use Unleash, would be nice to integrate our feature flags with Datadog via their [OpenFeature interation support](https://docs.datadoghq.com/feature_flags/).
Looking forward to it :)

**Comment by gastonfournier** [comment]
Announced in our 8.1 release notes: https://docs.getunleash.io/release-notes/2026/8/5 here we have the list of official Open Feature providers for Unleash: https://docs.getunleash.io/sdks#openfeature-providers

**Comment by gastonfournier** [comment]
We now have an officially supported list of providers: https://docs.getunleash.io/sdks#openfeature-providers and more community supported. As of now, we invite you to provide feedback either in our Slack: https://slack.unleash.run or by opening an issue in one of those repositories. 

Go build something great!

**Reply by askpt** [reply]
That is great news! I highly encourage to add those to the ecosystem list available here: https://openfeature.dev/ecosystem?instant_search%5BrefinementList%5D%5Bvendor%5D%5B0%5D=Unleash

**Reply by gastonfournier** [reply]
Thanks @askpt yes we have opened issues to request adding them in that list: https://github.com/open-feature/openfeature.dev/issues
